### ISSUE TITLE
API: category id type

### ISSUE THREAD

**Issue by jangernert** [Athou/commafeed#1139]
The ids of feeds & entries is i64 while categories are inconsistent. Sometimes its a string and other times i64. I feel like categories should also have an i64 id property.

I guess this exception was done to be able to represent the "all" category? If that's the only case, maybe `/category/get` should just return the top level categories & top level feeds? Or maybe reserve the id `0` or `-1` for "all"?

**Comment by Athou** [comment]
You're right it's inconsistent and it's indeed because there are the 'all' and 'starred' categories. 
Actually I feel like moving to a string id across the board instead of a long number would be the way to go, as it would prevent the need for special numeric values.

Probably for 4.x though since it would be a breaking change.
